### PR TITLE
[video_player] [WIP] Add web implementation

### PR DESCRIPTION
**Pull request by cbenhagen**
## Description
This is adding a web plugin for video_player using `HtmlElementView` and a generated plugin registrant.

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [ ] All existing and new tests are passing.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. [shared_preferences]
- [ ] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [ ] I updated CHANGELOG.md to add a description of the change.
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [x] No, this is *not* a breaking change.

### CODE REVIEW

**Comment by MateusAmin** [comment]
Thanks for working on this @cbenhagen  :D

File: packages/video_player/pubspec.yaml [hunk 1]
```diff
@@ -7,14 +7,23 @@ homepage: https://github.com/flutter/plugins/tree/master/packages/video_player
 
 flutter:
   plugin:
-    androidPackage: io.flutter.plugins.videoplayer
-    iosPrefix: FLT
-    pluginClass: VideoPlayerPlugin
+    platforms:
```
Comment (amirh) [inline packages/video_player/pubspec.yaml]: We can't land this before next stable release as the plugin won't compile with the stable tool.

**Comment by amirh** [comment]
This should be implemented as a federated package (see: https://flutter.dev/go/federated-plugins).

I think this plugin needs a refactoring to have a Dart platform interface before adding the web implementation.

cc @hterkelsen

**Comment by cbenhagen** [comment]
@amirh / @hterkelsen I feel a bit lost adding that platform interface and especially in adding the needed tests. Would you be able to provide some guidance / mentoring?

**Comment by hterkelsen** [comment]
Hi Ben, I'm happy to help! To get you started, you can take a look at the changes I made to `url_launcher` to add a platform interface:

* Adding the platform interface: https://github.com/flutter/plugins/pull/2217/files
* Using the interface in `package:url_launcher`: https://github.com/flutter/plugins/pull/2228/files
* Updating `url_launcher_web` to implement the platform interface: https://github.com/flutter/plugins/pull/2237/files

Please don't hesitate to ask if you have any questions

**Comment by cbenhagen** [comment]
@hterkelsen I used your changes as a template and opened https://github.com/flutter/plugins/pull/2273.

Is there an advantage in using method channels for the web plugins? Or should I just implement a WebVideoPlayer which does not use them like below?

```dart
class WebVideoPlayer extends VideoPlayerPlatform {
}
```

**Comment by cbenhagen** [comment]
Closing in favor of https://github.com/flutter/plugins/pull/2279

**Pull request closed without merging**